### PR TITLE
[sailfish-browser] Cleanup QMozContext references. Fixes JB#36498

### PR DESCRIPTION
**Pull request by rainemak**
Use WebEngine and WebEngineSettings when possible.

There's on include dependency in DeclarativeWebContainer to
QMozContext::TaskHandle. As posting tasks is rather low level API,
I don't see reason to duplicate that to WebEngine.

### CODE REVIEW

**Comment by rainemak** [comment]
@jpetrell / @chriadam review ?

File: src/core/declarativewebcontainer.cpp [hunk 1]
```diff
@@ -86,8 +87,10 @@ DeclarativeWebContainer::DeclarativeWebContainer(QWindow *parent)
     setTabModel(privateMode() ? m_privateTabModel.data() : m_persistentTabModel.data());
 
     connect(DownloadManager::instance(), SIGNAL(downloadStarted()), this, SLOT(onDownloadStarted()));
```
Comment (chriadam) [inline src/core/declarativewebcontainer.cpp]: could change this to the new connection syntax, for consistency

File: tests/auto/mocks/declarativewebutils/declarativewebutils.cpp [hunk 2]
```diff
@@ -21,7 +20,7 @@ DeclarativeWebUtils::DeclarativeWebUtils(QObject *parent)
     , m_homePage("file:///opt/tests/sailfish-browser/manual/testpage.html")
     , m_firstUseDone(true)
 {
-    connect(QMozContext::instance(), SIGNAL(onInitialized()),
+    connect(SailfishOS::WebEngineSettings::instance(), SIGNAL(initialized()),
```
Comment (chriadam) [inline tests/auto/mocks/declarativewebutils/declarativewebutils.cpp]: could use new connection syntax

File: tests/auto/tst_declarativewebcontainer/tst_declarativewebcontainer.cpp [hunk 3]
```diff
@@ -229,8 +229,6 @@ void tst_declarativewebcontainer::loading()
     QCOMPARE(m_webContainer->loading(), false);
 
     // Page is not loading => false
-    EXPECT_CALL(page, setWindow(m_webContainer.data()));
```
Comment (chriadam) [inline tests/auto/tst_declarativewebcontainer/tst_declarativewebcontainer.cpp]: is this related?

Comment (rainemak) [inline tests/auto/tst_declarativewebcontainer/tst_declarativewebcontainer.cpp]: Not per se... Just removed expected calls that are not longer happening. Will amend a comment to the commit log.

File: tests/auto/mocks/declarativewebutils/declarativewebutils.cpp (hunk 2 above)
Comment (rainemak) [inline tests/auto/mocks/declarativewebutils/declarativewebutils.cpp]: Fixed

File: src/core/declarativewebcontainer.cpp (hunk 1 above)
Comment (rainemak) [inline src/core/declarativewebcontainer.cpp]: Added a commit on top that fixes all connections.

**Comment by chriadam** [comment]
LGTM
